nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/FetchS3Object.java

```diff
@@ -148,6 +153,37 @@
         return problems;
     }
 
+    @Override
+    public List<ConfigVerificationResult> verify(ProcessContext context, ComponentLog verificationLogger, Map<String, String> attributes) {
+        final List<ConfigVerificationResult> results = new ArrayList<>(super.verify(context, verificationLogger, attributes));
+
+        final String bucket = context.getProperty(BUCKET).evaluateAttributeExpressions(attributes).getValue();
+        final String key = context.getProperty(KEY).evaluateAttributeExpressions(attributes).getValue();
+
+        final AmazonS3 client = getConfiguration(context).getClient();
+        final GetObjectRequest request = createGetObjectRequest(context, attributes);
+
+        try (final S3Object s3Object = client.getObject(request)) {
```

Comment:
Doing a full `GET` just to verify the processor seems a bit heavy handed (if it's a large Object then it might also take a while and be costly for the consumer both in terms of network bandwidth and the transfer fees from S3) - could we use a `HEAD` instead to check whether the Object exists?

Would that add to the S3 policies needed for a user to run this Processor though?

Reply:
Yes, so that's the downside -- the tradeoff between requiring different policies vs. a potentially heavy request.  So far the community has tended to err on the side of requiring exactly the same permissions.  Also, particularly with the Fetch processor, the user will be fetching an object they specifically provide in the attributes dialog, which gives them an opportunity not to specify a large object.

Reply:
A quick look at the [AWS docs](https://docs.aws.amazon.com/AmazonS3/latest/API/API_HeadObject.html) suggests that if one can `GetObject` then they will be able to `HeadObject`

Particularly as the user may need to specify a large object to be retrieved from S3 by the processor, doing this more times than is necessary seems like a waste and unnecessary extra cost. I know that my use of this Processor often results in pulling many objects that range from very small to much larger, I'd be wanting to turn this step off if it meant doubling up on pulling some of the larger files over the internet!

Reply:
Ah, I see what you mean with `HeadObject` -- that is what we want, then.